### PR TITLE
upgrade dependencies, including major upgrade to schema-utils

### PR DESCRIPTION
**Pull request by spalger**
Upgraded the dependencies to use latest versions, most were minor upgrades but `schema-utils` went up two major upgrades. According to the [`schema-utils` changelog](https://github.com/webpack/schema-utils/blob/master/CHANGELOG.md) the v1 -> v2 upgrade is the only time where code changes were necessary. The changes are very minor, simply adjusting the options to pass in the name and options prefix for the validation error. Minor changes to the istanbul libraries does trigger minor snapshot upgrades but they're ver broken on github, so for the reviewers sake here are the highlights handled better by local diff highlighters:
<details>
 <summary>terrible screenshot that highlights the changes in the snapshots</summary>
 
</details>

The upgraded package limit node version support to 8.9.0+ so I've included some pending changelog docs, upgraded the node versions being tested on CI, and upgraded to the latest version of npm to get support for node versions 12+.

Also migrated from `nps check` to `npm audit`, which should provide similar information, though I get the feeling that nobody is really looking at these results so maybe they can just be removed. There are a number of outdated packages with security vulnerabilities in the devDependencies, which should be fixed but felt out of the scope of this PR (I definitely tried to fix them) so I updated the audit command to `npm audit --production` to only check for security vulnerabilities in production dependencies.

Additionally, the CI checks run against `webpack@latest`, which require relatively minor webpack config changes to get tests passing.

### CODE REVIEW

**Comment by spalger** [comment]
Well, this has been a challenge. I'm not sure if what I've done here is too much for one PR, or if I have to get this passing against webpack canary.

I've gotten it close, but webpack canary is producing slightly different output and I'm really unsure what to do about that difference in the tests.

File: test/index.test.js [hunk 1]
```diff
@@ -1,33 +1,37 @@
+import path from 'path';
 import webpack from './utils/webpack';
 
 test('instrument code', async () => {
-  const stats = await webpack();
-  const instrumentedSource = stats.compilation.assets['main.js'].source();
+  const { stats, filesystem } = await webpack();
+
+  const instrumentedSource = filesystem.readFileSync(path.resolve(stats.compilation.options.output.path, 'main.js'), 'utf8');
```
Comment (spalger) [inline test/index.test.js:7]: This is necessary because stats won't include assets anymore in webpack 5.

**Comment by evilebottnawi** [comment]
@spalger Hi, we recently discussed to deprecate this loader, it is abandoned and not maintainable, so I did not expect a bit that someone would send PR, you can tell how you use the current loader, what is use cases?

**Comment by spalger** [comment]
Well, we currently use it to instrument our bundles before running them through a functional end-to-end testing suite. It's an experimental form of coverage collection and this works fine, we just had to force upgrade schema-utils with yarn resolutions and I was looking to remove a warning produced by yarn. Is there something intrinsically unmaintainable about this project that I'm not realizing? It seems like a very simple wrapper around istanbul that we could totally implement in our project rather than relying on the loader, but I'm also happy to help get this caught up to modern standards.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
 
+<a name="unreleased"></a>
+## unreleased
+
+
+### BREAKING CHANGES
+
+* Upgrades loader-utils, schema-utils, and istanbul-lib-instrument which all have a minimum NodeJS version of `>= 8.9.0`.
+
+
```
Comment (evilebottnawi) [inline CHANGELOG.md:13]: Remove this, we automatically generate it before release

File: package.json [hunk 3]
```diff
@@ -8,13 +8,13 @@
...
-    "loader-utils": "^1.1.0",
-    "schema-utils": "^0.3.0"
+    "convert-source-map": "^1.7.0",
+    "istanbul-lib-instrument": "^1.10.2",
+    "loader-utils": "^1.4.0",
+    "schema-utils": "^2.7.1"
   },
   "engines": {
-    "node": ">= 4.8 < 5.0.0 || >= 5.10"
+    "node": ">=8.9.0"
```
Comment (evilebottnawi) [inline package.json:17]: `10.13.0`, `8.9.0` is outdated

**Review by alexander-akait — COMMENTED** [review]
Also we need fix CI problems

**Comment by spalger** [comment]
> Also we need fix CI problems

This is tricky, as CI currently require that webpack@latest and webpack@master both create the same bundle as verified by the Jest snapshots. There are possibly things we could do to remove problematic bits from the snapshots with serializers, or by processing the file before snapshotting it, but I think I'm trying to understand the value of running the same test suite against master and latest, especially when the difference between the two is a major version break.

Is this how things work in other webpack-contrib projects?

File: .babelrc [hunk 4]
```diff
@@ -5,7 +5,7 @@
       {
         "useBuiltIns": true,
         "targets": {
-          "node": "4.8"
+          "node": "10"
```
Comment (evilebottnawi) [inline .babelrc:8]: Set `10.13` too

**Review by alexander-akait — APPROVED** [review]

**Comment by spalger** [comment]
Oh, cool, do you have plans to fix CI? I'm happy to work up the fixes but my instinct is to just drop the webpack@master tests. Do you have a preferred solution?

**Comment by alexander-akait** [comment]
@spalger 

> Oh, cool, do you have plans to fix CI?

Yes, today

> I'm happy to work up the fixes but my instinct is to just drop the webpack@master tests.

Yes, we should rewrite some tests

> Do you have a preferred solution?

I think yes, let's me try

**Pull request merged**